webapp/src/components/channel_header_actions/channel_header_menu.jsx

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import JiraIcon from 'components/icon';
+
+export default class ChannelHeaderMenuAction extends React.PureComponent {
+    static propTypes = {
+        open: PropTypes.func.isRequired,
+        channelId: PropTypes.string.isRequired,
+        isInstanceInstalled: PropTypes.bool.isRequired,
+        userConnected: PropTypes.bool.isRequired,
+    };
+
+    handleClick = () => {
+        const {open, channelId, userConnected} = this.props;
+        if (this.props.isInstanceInstalled && userConnected) {
+            open(channelId);
+        }
```

Comment:
Seems weird to deconstruct all but one prop. I think this order is clearer as well, since the first two I have here are the most significant. 0/5 on that part

```suggestion
        const {isInstanceInstalled, userConnected, open, channelId} = this.props;
        if (isInstanceInstalled && userConnected) {
            open(channelId);
        }
```